controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -426,13 +407,66 @@ func (r *ROSAControlPlaneReconciler) rosaClusterToROSAControlPlane(log *logger.L
 	}
 }
 
-func (r *ROSAControlPlaneReconciler) getOcmCluster(rosaScope *scope.ROSAControlPlaneScope, ocmConnection *sdk.Connection) (*cmv1.Cluster, error) {
-	clusterKey := rosaScope.ControlPlane.Name
+// OCMClient is a temporary helper to talk to OCM API.
+// TODO(alberto): vendor this from https://github.com/openshift/rosa/tree/master/pkg/ocm or build its own package here.
+type OCMClient struct {
+	ocm *sdk.Connection
+}
+
+func newOCMClient() (*OCMClient, error) {
+	// Create the connection, and remember to close it:
+	token := os.Getenv("OCM_TOKEN")
+	ocmAPIUrl := os.Getenv("OCM_API_URL")
+	if ocmAPIUrl == "" {
+		ocmAPIUrl = "https://api.openshift.com"
+	}
+
+	// Create a logger that has the debug level enabled:
+	ocmLogger, err := sdk.NewGoLoggerBuilder().
+		Debug(false).
+		Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build logger: %w", err)
+	}
+
+	connection, err := sdk.NewConnectionBuilder().
+		Logger(ocmLogger).
+		Tokens(token).
+		URL(ocmAPIUrl).
+		Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to ocm client: %w", err)
+	}
+	ocmClient := OCMClient{ocm: connection}
+
+	return &ocmClient, nil
+}
+
+func (client *OCMClient) Close() error {
+	return client.ocm.Close()
+}
+
+func (client *OCMClient) CreateCluster(clusterSpec *cmv1.Cluster) (*cmv1.Cluster, error) {
```

Comment:
Do we really need to build a further abstraction on top of the existing client SDK?